corehq/apps/enterprise/decorators.py

```diff
@@ -0,0 +1,12 @@
+from functools import wraps
+
+from corehq.apps.accounting.utils.subscription import get_account_or_404
+
+
+def require_enterprise_admin(view_func):
+    @wraps(view_func)
+    def _inner(request, domain, *args, **kwargs):
+        # todo separate authorization from get method
+        request.account = get_account_or_404(request, domain)
```

Comment:
i did not name the original function. i am simply adding it to the decorator and will clean it up afterward. i agree it is poorly named. the authorization logic should happen in the decorator, not in a `get`

Reply:
I think I understand -- when you say clean up, you mean that it will likely return a different HTTP code than 404?

Reply:
I think `get_account_or_404` makes sense...`404` vs `403` is fine here...because it's not checking permissions but the existence of the account. However, the **permissions** check for whether the current user has enterprise admin privileges should absolutely not live there! Our pattern is to do permissions checks in decorators, and this clearly does not follow that pattern

Reply:
see this PR for refactor. going to leave it open for review and trigger QA next week https://github.com/dimagi/commcare-hq/pull/29207